mm2src/coins/lightning.rs

```diff
@@ -15,6 +26,10 @@ mod ln_errors;
 mod ln_rpc;
 #[cfg(not(target_arch = "wasm32"))] mod ln_utils;
 
+lazy_static! {
+    static ref REQUEST_IDX: Arc<AtomicU64> = Arc::new(AtomicU64::new(0));
```

Comment:
It might cause problems actually. I guess the request_id has to be persisted in storage too.
I will have to inspect LDK also for what happens if a node is restarted before [`SendAcceptChannel`](https://docs.rs/lightning/0.0.101/lightning/util/events/enum.MessageSendEvent.html#variant.SendAcceptChannel) msg is received and `FundingGenerationReady` event is fired. Does the other node replay the `SendAcceptChannel` msg or does the channel gets discarded before the funding transaction.

Reply:
The request_id (user_channel_id) is also needed for the `ChannelClosed` event so I guess persisting it in storage is the solution. One case for the `ChannelClosed` event is that it will be fired if the other peer is disconnected before the funding transaction was broadcasted, which means if our node was restarted before sending the funding transaction and we couldn't re-establish the connection with the other channel peer, we will need the request_id for the `ChannelClosed` event, therefore, the request_id needs to be persisted.